tests/test_harness.rs

```diff
@@ -0,0 +1,80 @@
+use std::{
+    env,
+    fs::canonicalize,
+    io::{Error, ErrorKind},
+    process::{Command, ExitStatus},
+};
+
+pub struct Harness;
+
+type Result<T = ()> = std::io::Result<T>;
+
+/// Maps an exit status to a result, meaning that a non-successful exit status
+/// will get mapped to an `Err` variant
+trait ExitStatusConverter {
+    fn into_result(self) -> Result;
+}
+
+impl ExitStatusConverter for Result<ExitStatus> {
+    fn into_result(self) -> Result {
+        match self {
+            Ok(exit_code) => match exit_code.success() {
+                true => Ok(()),
+                false => Err(Error::new(
+                    ErrorKind::Other,
+                    "command did not exit successfully",
+                )),
+            },
+            Err(e) => Err(e),
+        }
+    }
+}
+
+impl Harness {
+    /// Build the project present in the current directory using `rustc` or `gccrs`
+    fn cargo_build(use_gccrs: bool) -> Result {
+        let mut cmd = Command::new("cargo");
+
+        if use_gccrs {
+            // Add `gccrs` argument so that `cargo build` becomes `cargo gccrs build`
+            cmd.arg("gccrs");
+
+            // Tweak the path so that the most recently compiled *debug* version of
+            // cargo-gccrs is available as a subcommand
+            let path = env::var("PATH").unwrap();
+
+            cmd.env(
+                "PATH",
+                format!(
+                    "{}:{}",
```

Comment:
If this is to work on Windows, it will need to use [`env::join_paths`](https://doc.rust-lang.org/stable/std/env/fn.join_paths.html). The docs contain an example on how to append to the path.

Reply:
Thanks! I didn't know about this!

Reply:
Fixed in d68f0d7